test/unit/gateways/ebanx_test.rb

```diff
@@ -54,6 +54,28 @@ def test_successful_purchase_with_soft_descriptor
     assert_success response
   end
 
+  def test_successful_purchase_without_merchant_payment_code
+    response = stub_comms(@gateway, :ssl_request) do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_method, _endpoint, data, _headers|
+      assert_match %r{"merchant_payment_code\":\"1\"}, data
```

Comment:
I just realized that this assertion is being satisfied by the override in the metadata - can you also add an `assert_match` for both tests that shows we are sending the correct hexdigest value for the value that's nested under `payment`?